### PR TITLE
Remove generate-docs step in release

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.

### Summary of changes
This pr contains the following changes:
- Output the snapshot name if a snapshot is uploaded to registry to enable us to use the snapshot easily.
- As we now generate and create the pr for docs locally,  the `generate-docs` step is not needed, so delete this stop from release steps

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -195,6 +195,7 @@ commands:
                   if [[ -n ${CIRCLE_BRANCH+x} ]] && [[ $CIRCLE_BRANCH != "main" ]] && [[ "$(git log -1)" = *"publish_android_snapshot"* ]]; then
                     sed -i -e "s/^VERSION_NAME=.*/VERSION_NAME=${LAST_RELEASE:9}-${CIRCLE_BRANCH}-SNAPSHOT/" gradle.properties
                     curl -Ls https://mapbox-release-engineering.s3.amazonaws.com/mbx-ci/latest/mbx-ci-linux-amd64 > mbx-ci && chmod 755 ./mbx-ci
+                    head -n 1 gradle.properties
```
Comment (kiryldz) [inline .circleci/config.yml:198]: Why do we need it?

Comment (kiryldz) [reply .circleci/config.yml:198]: If I understand correctly we just print first line from `gradle.properties` - is it used somehow afterwards?

**Review by kiryldz — APPROVED** [review]
LGTM with question

File: .circleci/config.yml (hunk 1 above)
Comment (Chaoba) [reply .circleci/config.yml:198]: If you build a snapshot, you can easy copy the version name in CI and paste in your project

Comment (pengdev) [reply .circleci/config.yml:198]: I think `make sdkRegistryUpload` task will also print the artifact name of the snapshot.

Comment (Chaoba) [reply .circleci/config.yml:198]: Oh right, haven't seen that before.

**Pull request merged**